# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/CanvasDrawEventRecorder.h
# Commit: 31935f164ef0
# Full Hash: 31935f164ef0d6c4668f211482c139847dcab7f2
# Author: Lee Salzman <lsalzman@mozilla.com>
# Date: 2023-12-13 00:43:26
# Regressor Bug: 1829026
# File Overlap Count: 1
# Description:
#   Bug 1829026 - Update CanvasTranslator to work with DrawTargetWebgl. r=aosmond
#   
#   This adds the necessary infrastructure for CanvasTranslator to allocate DrawTargetWebgl
#   instead of just allocating TextureData, and to use RemoteTextureMap to handle sending
#   the DrawTargetWebgl frames to the compositor.
# ==============================================================================

diff -r 13c9bf69085f -r 31935f164ef0 gfx/layers/CanvasDrawEventRecorder.h
--- a/gfx/layers/CanvasDrawEventRecorder.h	Tue Dec 12 07:35:02 2023 +0000
+++ b/gfx/layers/CanvasDrawEventRecorder.h	Tue Dec 12 07:35:03 2023 +0000
@@ -67,16 +67,13 @@
    public:
     virtual ~Helpers() = default;
 
-    virtual bool InitTranslator(const TextureType& aTextureType,
-                                Handle&& aReadHandle,
-                                nsTArray<Handle>&& aBufferHandles,
-                                const uint64_t& aBufferSize,
-                                CrossProcessSemaphoreHandle&& aReaderSem,
-                                CrossProcessSemaphoreHandle&& aWriterSem,
-                                const bool& aUseIPDLThread) = 0;
+    virtual bool InitTranslator(
+        TextureType aTextureType, gfx::BackendType aBackendType,
+        Handle&& aReadHandle, nsTArray<Handle>&& aBufferHandles,
+        uint64_t aBufferSize, CrossProcessSemaphoreHandle&& aReaderSem,
+        CrossProcessSemaphoreHandle&& aWriterSem, bool aUseIPDLThread) = 0;
 
-    virtual bool AddBuffer(Handle&& aBufferHandle,
-                           const uint64_t& aBufferSize) = 0;
+    virtual bool AddBuffer(Handle&& aBufferHandle, uint64_t aBufferSize) = 0;
 
     /**
      * @returns true if the reader of the CanvasEventRingBuffer has permanently
@@ -90,7 +87,8 @@
     virtual bool RestartReader() = 0;
   };
 
-  bool Init(TextureType aTextureType, UniquePtr<Helpers> aHelpers);
+  bool Init(TextureType aTextureType, gfx::BackendType aBackendType,
+            UniquePtr<Helpers> aHelpers);
 
   /**
    * Record an event for processing by the CanvasParent's CanvasTranslator.